src/plus/integrations/integration.ts

```diff
@@ -362,6 +362,31 @@ export abstract class IntegrationBase<
 		resource: T,
 		id: string,
 	): Promise<IssueOrPullRequest | undefined>;
+
+	async getCurrentAccount(options?: { avatarSize?: number }): Promise<Account | undefined> {
+		const scope = getLogScope();
+
+		const connected = this.maybeConnected ?? (await this.isConnected());
+		if (!connected) return undefined;
+
+		const currentAccount = this.container.cache.getCurrentAccount(this, () => ({
+			value: (async () => {
+				try {
+					const account = await this.getProviderCurrentAccount(this._session!, options);
+					this.resetRequestExceptionCount();
+					return account;
+				} catch (ex) {
+					return this.handleProviderException<Account | undefined>(ex, scope, undefined);
+				}
+			})(),
+		}));
+		return currentAccount;
+	}
+
+	protected abstract getProviderCurrentAccount(
```

Comment:
We can make this optional rather than `abstract` so that the providers that don't support it, don't need a dummy implementation.

I had an example of that in the Cloud Patch I sent with the PR search changes.